src/app/service/statistics.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+import { IpfsService } from './ipfs.service';
+import { SidebarService } from './sidebar.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class StatisticsService {
+    private runStatistics = false;
+    
+    constructor(
+        private ipfs: IpfsService,
+        public sidebar: SidebarService
+    ) {
+        sidebar.topic.subscribe(async whatToShow => {
+            this.runStatistics = (whatToShow === 'statistics');
+            if (this.runStatistics) {
+                this.updateStatsBw();
+            }
+        });
+    }
+
+    public statsBw = null;
+    private async updateStatsBw() {
+        while (this.runStatistics) {
```

Comment:
Just a nitpick, but the JS way would be
`intervalId = setInteval(cb, 2000)` 
for this kind of task. An then stop via 
`clearInverval(intervalId)`

Reply:
thanks, that does help! will try again